storage.go

```diff
@@ -0,0 +1,342 @@
+	AwsEndpoint         string
+	AwsRegion           string
+	AwsDisableSSL       bool
+	LockTimeout         time.Duration
+	LockPollingInterval time.Duration
+}
+
+// initConfig initializes configuration for table nam and AWS session
+func (s *Storage) initConfig() error {
+	if s.Table == "" {
+		return fmt.Errorf("config error: table name is required")
+	}
+
+	if s.LockTimeout == 0*time.Second {
+		s.LockTimeout = lockTimeoutMinutes
+	}
+	if s.LockPollingInterval == 0*time.Second {
+		s.LockPollingInterval = lockPollingInterval
+	}
+
+	// Initialize AWS Session if needed
+	if s.AwsSession == nil {
+		var err error
+		s.AwsSession, err = session.NewSession(&aws.Config{
+			Endpoint:   &s.AwsEndpoint,
+			Region:     &s.AwsRegion,
+			DisableSSL: &s.AwsDisableSSL,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Store puts value at key.
+func (s *Storage) Store(key string, value []byte) error {
+	if err := s.initConfig(); err != nil {
+		return err
+	}
+
+	encVal := base64.StdEncoding.EncodeToString(value)
+
+	if key == "" {
+		return fmt.Errorf("key must not be empty")
+	}
+
+	svc := dynamodb.New(s.AwsSession)
+	input := &dynamodb.PutItemInput{
+		Item: map[string]*dynamodb.AttributeValue{
+			primaryKeyAttribute: {
+				S: aws.String(key),
+			},
+			contentsAttribute: {
+				S: aws.String(encVal),
+			},
+			lastUpdatedAttribute: {
+				S: aws.String(time.Now().Format(time.RFC3339)),
+			},
+		},
+		TableName: aws.String(s.Table),
+	}
+
+	_, err := svc.PutItem(input)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Load retrieves the value at key.
+func (s *Storage) Load(key string) ([]byte, error) {
+	if err := s.initConfig(); err != nil {
+		return []byte{}, err
+	}
+
+	if key == "" {
+		return []byte{}, fmt.Errorf("key must not be empty")
+	}
+
+	domainItem, err := s.getItem(key)
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(domainItem.Contents), nil
+}
+
+// Delete deletes key.
+func (s *Storage) Delete(key string) error {
+	if err := s.initConfig(); err != nil {
+		return err
+	}
+
+	if key == "" {
+		return fmt.Errorf("key must not be empty")
+	}
+
+	svc := dynamodb.New(s.AwsSession)
+	input := &dynamodb.DeleteItemInput{
+		Key: map[string]*dynamodb.AttributeValue{
+			primaryKeyAttribute: {
+				S: aws.String(key),
+			},
+		},
+		TableName: aws.String(s.Table),
+	}
+
+	_, err := svc.DeleteItem(input)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Exists returns true if the key exists
+// and there was no error checking.
+func (s *Storage) Exists(key string) bool {
+
+	cert, err := s.Load(key)
+	if string(cert) != "" && err == nil {
+		return true
+	}
+
+	return false
+}
+
+// List returns all keys that match prefix.
+// If recursive is true, non-terminal keys
+// will be enumerated (i.e. "directories"
+// should be walked); otherwise, only keys
+// prefixed exactly by prefix will be listed.
+func (s *Storage) List(prefix string, recursive bool) ([]string, error) {
```

Comment:
`recursive` isn't used. Is it a placeholder for future functionality, or a mistake?

Reply:
it's required as part of the interface but seems to only apply to filesystem type storage methods, I don't know how it would be used in this case so i left it unused

Reply:
OK, maybe would be good to say that in the comment.